### PR TITLE
Futurepass migration extrinsic removal

### PR DESCRIPTION
**Pull request by zees-dev**
## Changelog

- Removed futurepass migration extrinsic (deprecated and unused)
  - This should reduce the runtime WASM size for futurepass pallet
- Added migration to purge on-chain storage item (migrator admin account)
- Updated mocks and tests accordingly
- Fixed bug with futurepass extrinsics not being benchmarkable (build failed with `runtime-benchmarks` flag)

### CODE REVIEW

File: pallet/futurepass/src/lib.rs [hunk 1]
```diff
@@ -178,10 +157,6 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type DefaultProxy<T: Config> = StorageMap<_, Twox64Concat, T::AccountId, T::AccountId>;
 
-	/// Migration data for user (root) and collections they can migrate
-	#[pallet::storage]
-	pub type MigrationAdmin<T: Config> = StorageValue<_, T::AccountId, OptionQuery>;
```
Comment (surangap) [inline pallet/futurepass/src/lib.rs:183]: nit - remove the existing value. kill the storage manually or write a small cleanup migration.

**Review by surangap — APPROVED** [review]
lgtm

File: pallet/futurepass/src/lib.rs (hunk 1 above)
Comment (zees-dev) [reply pallet/futurepass/src/lib.rs:183]: Ah yep good point; will look into simply destroying the storage value without a migration.
Might need sudo access for that.

**Review by JasonTulp — APPROVED** [review]
Lgtm!

**Comment by zees-dev** [comment]
@JasonTulp, @surangap - updated PR to incl. migration of storage item removal

File: runtime/src/migrations/futurepass.rs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		let pallet_prefix = Twox128::hash(b"Futurepass");
+		let storage_item_name = Twox128::hash(b"MigrationAdmin");
+		let storage_key = [pallet_prefix, storage_item_name].concat();
+
+		// Remove the storage item
+		storage::clear(&storage_key);
+
+		log::info!(target: "Migration", "Futurepass: successfully killed MigrationAdmin storage item");
+
+		Weight::from_ref_time(1u64)
```
Comment (JasonTulp) [inline runtime/src/migrations/futurepass.rs:110]: Does clearing a storage count as a write in terms of weight?

**Review by JasonTulp — APPROVED** [review]
lgtm! Nice work

**Review by surangap — APPROVED** [review]

File: runtime/src/migrations/futurepass.rs (hunk 2 above)
Comment (zees-dev) [reply runtime/src/migrations/futurepass.rs:110]: Yep I believe so (since you'd be nullifying it).

**Pull request merged**